esp/esp/users/models/__init__.py

```diff
@@ -127,62 +122,23 @@ def save(self, *args, **kwargs):
-    This user extends the auth.User of django"""
+class BaseESPUser(object):
+    """ Base class for ESPUser and AnonymousESPUser.
+    Pretty much anything from ESPUser that isn't directly related
+    to being a model should go here. """
 
-    class Meta:
-        proxy = True
-        verbose_name = 'ESP User'
-        
     objects = ESPUserManager()
-    # this will allow a casting from User to ESPUser:
-    #      foo = ESPUser(bar)   <-- foo is now an ``ESPUser''
-    def __init__(self, userObj=None, *args, **kwargs):
-        # Set up the storage for instance state
-        self._state = ModelState()
-    
-        # A bit of a hack: if we're passed a SimpleLazyObject, make sure it's
-        # initialized (with `dir` which is proxied), and then grab the __dict__
-        # of the wrapped User or ESPUser.   This is necessary because
-        # SimpleLazyObject doesn't proxy __dict__, so copying it would fail.
-        if isinstance(userObj, SimpleLazyObject):
-            dir(userObj)
-            self.__dict__ = userObj._wrapped.__dict__
-            # This is just a method, so SimpleLazyObj will proxy it just fine.
-            self._is_anonymous = userObj.is_anonymous()
-
-        # TODO(benkraft): in the case of an ESPUser, we should consider
-        # overriding __new__ to just return the ESPUser it was passed; I don't
-        # know why you'd call ESPUser on an ESPUser except by accident, but
-        # giving you back your ESPUser should work just fine.  On the other
-        # hand, this might be trickier than it sounds because there are a bunch
-        # of metaclasses flying around, and because if __new__ returns an
-        # ESPUser it will then get __init__ called on it, which might be
-        # unnecessary.
-        elif isinstance(userObj, (ESPUser, User, AnonymousUser)):
-            self.__dict__ = userObj.__dict__
-            self._is_anonymous = userObj.is_anonymous()
-
-        elif userObj is not None or len(args) > 0:
-            # Initializing a model using non-keyworded args is a horrible idea.
-            # No clue why you'd do it, but I won't stop you. -ageng 2009-05-10
-            User.__init__(self, userObj, *args, **kwargs)
-            self._is_anonymous = False
-        else:
-            User.__init__(self, *args, **kwargs)
-            self._is_anonymous = False
-
-        if not hasattr(self, "_state"):
-            ## Django doesn't properly insert this field on proxy models, apparently?
-            ## So, fake it. -- aseering 6/28/2010
-            self._state = FakeState()
-
-        self.other_user = False
+    other_user = False
 
+    def __init__(self, *args, **kwargs):
+        # This is last in ESPUser's method resolution order, and
+        # AnonymousUser doesn't do anything in its __init__,
+        # so there's no need for a super() call unless you're changing
+        # inheritance structure of this, ESPUser, or AnonymousESPUser,
+        # or if Django has changed something.
```

Comment:
It would be a bit weird, because if I'm just including it for safety it's unclear whether the correct thing to write is
`super(BaseESPUser, self).__init__(*args, **kwargs)`
or
`super(BaseESPUser, self).__init__()`

The first version involves trying to pass arguments to `AnonymousUser.__init__()`, which will break if there's ever anything actually there, and that feels ugly. The second version seems about as vulnerable to breaking as just leaving it out.

That's not actually the original reason, though, the original reason is just that I got really excited about having solved the multiple inheritance puzzle.